hawkular-alerts-engine/src/main/resources/org/hawkular/alerts/engine/rules/ConditionMatch.drl

```diff
@@ -260,6 +264,33 @@ rule Rate
         insert( ce );
 end
 
+rule UpdateMissingStateFromData
+    when
+        $md : MissingState( $tenantId : tenantId, $tsource : source, $did : dataId, previousTime < time )

```

Comment:
perhaps rename this variable from $md to $ms, that is sort of the convention we have.